File: bitget-php-sdk-api/src/api/v2/MixOrderApi.php
<?php


namespace bitget\api\v2;


use bitget\internal\BitgetApiClient;

class MixOrderApi
{
    var BitgetApiClient $BitgetApiClient;

    public function __construct($BitgetApiClient)
    {
        $this->BitgetApiClient = $BitgetApiClient;
    }

    // normal order
    public function placeOrder($params): string
    {
        return $this->BitgetApiClient->doPost("/api/v2/mix/order/place-order", $params);
    }

    public function batchPlaceOrder($params): string
    {
        return $this->BitgetApiClient->doPost("/api/v2/mix/order/batch-place-order", $params);
    }

    public function cancelOrder($params): string
    {
        return $this->BitgetApiClient->doPost("/api/v2/mix/order/cancel-order", $params);
    }

    public function batchCancelOrders($params): string
    {
        return $this->BitgetApiClient->doPost("/api/v2/mix/order/batch-cancel-orders", $params);
    }

    public function ordersHistory($params): string
    {
        return $this->BitgetApiClient->doGet("/api/v2/mix/order/orders-history", $params);
    }

    public function ordersPending($params): string
    {
        return $this->BitgetApiClient->doGet("/api/v2/mix/order/orders-pending", $params);
    }

    public function fills($params): string
    {
        return $this->BitgetApiClient->doGet("/api/v2/mix/order/fills", $params);
    }


    // plan
    public function placePlanOrder($params): string
    {
        return $this->BitgetApiClient->doPost("/api/v2/mix/order/place-plan-order", $params);
    }

    public function cancelPlanOrder($params): string
    {
        return $this->BitgetApiClient->doPost("/api/v2/mix/order/cancel-plan-order", $params);
    }

    public function ordersPlanPending($params): string
    {
        return $this->BitgetApiClient->doGet("/api/v2/mix/order/orders-plan-pending", $params);
    }

    public function ordersPlanHistory($params): string
    {
        return $this->BitgetApiClient->doGet("/api/v2/mix/order/orders-plan-history", $params);
    }


    // trader
    public function traderOrderClosePositions($params): string
    {
        return $this->BitgetApiClient->doPost("/api/v2/copy/mix-trader/order-close-positions", $params);
    }

    public function traderOrderCurrentTrack($params): string
    {
        return $this->BitgetApiClient->doGet("/api/v2/copy/mix-trader/order-current-track", $params);
    }

    public function traderOrderHistoryTrack($params): string
    {
        return $this->BitgetApiClient->doGet("/api/v2/copy/mix-trader/order-history-track", $params);
    }

    public function followerClosePositions($params): string
    {
        return $this->BitgetApiClient->doPost("/api/v2/copy/mix-follower/close-positions", $params);
    }

    public function followerQueryCurrentOrders($params): string
    {
        return $this->BitgetApiClient->doGet("/api/v2/copy/mix-follower/query-current-orders", $params);
    }

    public function followerQueryHistoryOrders($params): string
    {
        return $this->BitgetApiClient->doGet("/api/v2/copy/mix-follower/query-history-orders", $params);
    }

}

File: bitget-java-sdk-api/src/main/java/com/bitget/openapi/api/v2/MixOrderApi.java
package com.bitget.openapi.api.v2;

import com.bitget.openapi.dto.response.ResponseResult;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.QueryMap;

import java.util.Map;

public interface MixOrderApi {

    // normal order
    @POST("/api/v2/mix/order/place-order")
    Call<ResponseResult> placeOrder(@Body Map<String, String> paramMap);

    @POST("/api/v2/mix/order/batch-place-order")
    Call<ResponseResult> batchPlaceOrder(@Body Map<String, String> paramMap);

    @POST("/api/v2/mix/order/cancel-order")
    Call<ResponseResult> cancelOrder(@Body Map<String, String> paramMap);

    @POST("/api/v2/mix/order/batch-cancel-orders")
    Call<ResponseResult> batchCancelOrders(@Body Map<String, String> paramMap);

    @GET("/api/v2/mix/order/orders-history")
    Call<ResponseResult> ordersHistory(@QueryMap Map<String, String> paramMap);

    @GET("/api/v2/mix/order/orders-pending")
    Call<ResponseResult> ordersPending(@QueryMap Map<String, String> paramMap);

    @GET("/api/v2/mix/order/fills")
    Call<ResponseResult> fills(@QueryMap Map<String, String> paramMap);


    // plan
    @POST("/api/v2/mix/order/place-plan-order")
    Call<ResponseResult> placePlanOrder(@Body Map<String, String> paramMap);

    @POST("/api/v2/mix/order/cancel-plan-order")
    Call<ResponseResult> cancelPlanOrder(@Body Map<String, String> paramMap);

    @GET("/api/v2/mix/order/orders-plan-pending")
    Call<ResponseResult> ordersPlanPending(@QueryMap Map<String, String> paramMap);

    @GET("/api/v2/mix/order/orders-plan-history")
    Call<ResponseResult> ordersPlanHistory(@QueryMap Map<String, String> paramMap);


    // trader
    @POST("/api/v2/copy/mix-trader/order-close-positions")
    Call<ResponseResult> traderOrderClosePositions(@Body Map<String, String> paramMap);

    @GET("/api/v2/copy/mix-trader/order-current-track")
    Call<ResponseResult> traderOrderCurrentTrack(@QueryMap Map<String, String> paramMap);

    @GET("/api/v2/copy/mix-trader/order-history-track")
    Call<ResponseResult> traderOrderHistoryTrack(@QueryMap Map<String, String> paramMap);

    @POST("/api/v2/copy/mix-follower/close-positions")
    Call<ResponseResult> followerClosePositions(@Body Map<String, String> paramMap);

    @GET("/api/v2/copy/mix-follower/query-current-orders")
    Call<ResponseResult> followerQueryCurrentOrders(@QueryMap Map<String, String> paramMap);

    @GET("/api/v2/copy/mix-follower/query-history-orders")
    Call<ResponseResult> followerQueryHistoryOrders(@QueryMap Map<String, String> paramMap);
}
